contrib/fedora/rpm/build.sh

```diff
@@ -0,0 +1,329 @@
+
+UUID="$(uuidgen)"
+FW_BUILD_USERNAME="${FW_BUILD_USERNAME:-"$(git config user.name) <$(git config user.email)>"}" || die "git directory not properly setup or working?"
+CHANGELOG="$CHANGELOG"
+VERSION="$(grep "^Version:" ./firewalld.spec |cut -f2 -d ' ')" || die "Failure to detect version from firewalld.spec"
+RELEASE_VERSION="$(git rev-list HEAD | wc -l)" || die "Failure to detect RELEASE_VERSION from git"
+SNAPSHOT="${SNAPSHOT:-%{nil\}}"
+COMMIT_FULL="$(git rev-parse --verify HEAD)" || die "Error reading HEAD revision"
+COMMIT="$(printf '%s' "$COMMIT_FULL" | sed 's/^\(.\{10\}\).*/\1/')"
+RPMBUILD_ARGS="$RPMBUILD_ARGS"
+
+LOG "conf: BASEDIR=\"$BASEDIR\""
+LOG "conf: DATE=\"$DATE\""
+LOG "conf: UUID=\"$UUID\""
+LOG "conf: CHANGELOG=\"$CHANGELOG\""
+LOG "conf: VERSION=\"$VERSION\""
+LOG "conf: RELEASE_VERSION=\"$RELEASE_VERSION\""
+LOG "conf: COMMIT_FULL=\"$COMMIT_FULL\""
+LOG "conf: COMMIT=\"$COMMIT\""
+LOG "conf: FORCE=\"$ARG_FORCE\""
+LOG "conf: BUILDTYPE=\"$BUILDTYPE\""
+LOG "conf: RPMBUILD_ARGS=\"$RPMBUILD_ARGS\""
+LOG "conf: ARGS_WITH=\"${ARGS_WITH[@]}\""
+LOG "conf: ARGS_WITHOUT=\"${ARGS_WITHOUT[@]}\""
+LOG "conf: FW_BUILD_USERNAME=\"$FW_BUILD_USERNAME\""
+LOG "conf: FW_BUILD_SNAPSHOT=\"$FW_BUILD_SNAPSHOT\""
+
+TARBALL_NAME="firewalld-$VERSION.tar.bz2"
+CALL_DIST=0
+
+if [ "$TARBALL" = "" -a "$ARG_FORCE" != 1 -a -f "./$TARBALL_NAME" ]; then
+    TARBALL="./$TARBALL_NAME"
+fi
+
+if [ "$TARBALL" != "" ]; then
+    :
+else
+    if [ "$ARG_FORCE" = 1 ]; then
+        git clean -fdx -- ":(exclude)$RPMDIR/" ":(exclude)firewalld-*.tar.bz"
+    else
+        if [ "$(git clean -ndx -- ":(exclude)$RPMDIR/" ":(exclude)firewalld-*.tar.bz")" != "" ] ; then
+            die "Working directory is not clean. Rerun with --force. WARNING: this cleans the working directory!"
+        fi
+    fi
+    CALL_DIST=1
+fi
+
+BUILDDIR="$(mktemp -d -p "$RPMDIR" "Builddir.$DATE.XXXXX")"
+
+LOG "conf: BUILDDIR=\"$BUILDDIR\""
+
+ln -snf "$(basename "$BUILDDIR")" "$RPMDIR/Builddir.building"
+mkdir -p "$BUILDDIR/SPECS/"
+mkdir -p "$BUILDDIR/SOURCES/"
+
+ln "$BUILDLOG" "$BUILDDIR/build.log"
+rm -f "$BUILDLOG"
+
+SPECFILE="$BUILDDIR/SPECS/firewalld.spec"
+
+dist() {
+    ./autogen.sh
+    ./configure
+    make dist
+    TARBALL="./$TARBALL_NAME"
+}
+
+if [ "$CALL_DIST" = "1" ]; then
+    dist
+fi
+
+if [ ! -f "$TARBALL" -o "$(printf "%s" "$TARBALL" | wc -w)" -ne 1 ] ; then
+    if [ "$CALL_DIST" = "1" ]; then
+        die "Tarball not generated by \`make dist\`. Something went wrong."
+    fi
+    die "To tarball found. Build with --force (warning: cleans git tree)"
+fi
+
+write_changelog() {
+    if [ -z "$CHANGELOG" ]; then
+        cat <<- EOF
+	* $(LC_TIME=C date '+%a %b %d %Y') $FW_BUILD_USERNAME - %{epoch_version}:%{version}-%{release_version}%{?snap}
+	- build of firewalld ($DATE, uuid: $UUID, git: $COMMIT_FULL)
+	$(git log -n20 --date=local --format='- %h %s [%an] (%ci)')
+	- ...
+	EOF
+    else
+        echo "$CHANGELOG"
+    fi > "$BUILDDIR/SOURCES/CHANGELOG"
+}
```

Comment:
Please omit changelog support. spec file changelogs are supposed to be about _packaging_. It does not make sense to fill them with the git history.